u8timeseries/timeseries.py

```diff
@@ -20,35 +19,34 @@
 
 class TimeSeries:
     def __init__(self,
-                 series: pd.Series,
-                 confidence_lo: Optional[pd.Series] = None,
-                 confidence_hi: Optional[pd.Series] = None,
+                 series: Union[pd.Series, pd.DataFrame],
                  freq: Optional[str] = None,
                  fill_missing_dates: Optional[bool] = True):
         """
-        A TimeSeries is an object representing a univariate time series, and optional confidence intervals.
+        A TimeSeries is an object representing a time series.
 
         TimeSeries are meant to be immutable.
 
         Parameters
         ----------
         series
-            The actual time series, as a pandas Series with a proper time index.
-        confidence_lo
-            Optionally, a Pandas Series representing lower confidence interval.
-        confidence_hi
-            Optionally, a Pandas Series representing upper confidence interval.
+            The actual time series, as a pandas Series or DataFrame with a proper time index.
         freq
             Optionally, a string representing the frequency of the Pandas Series. When creating a TimeSeries
             instance with a length smaller than 3, this argument must be passed.
         fill_missing_dates
             Optionally, a boolean value indicating whether to fill missing dates with NaN values
             in case the frequency of `series` cannot be inferred.
         """
+        # create dataframes
+        if (isinstance(series, pd.Series)):
+            series = pd.DataFrame(series)
+        series.columns = range(series.shape[1])
 
-        raise_if_not(len(series) > 0, 'Series must not be empty.', logger)
+        raise_if_not(len(series) > 0 and series.shape[1] > 0, 'Series must not be empty.', logger)
         raise_if_not(isinstance(series.index, pd.DatetimeIndex), 'Series must be indexed with a DatetimeIndex.', logger)
-        raise_if_not(np.issubdtype(series.dtype, np.number), 'Series must contain numerical values.', logger)
+        raise_if_not(series.dtypes.apply(lambda x: np.issubdtype(x, np.number)).all(), 'Series must'
+                     ' contain only numerical values.', logger)
         raise_if_not(len(series) >= 3 or freq is not None, 'Series must have at least 3 values if the "freq" argument'
                      'is not passed', logger)
 
```

Comment:
Just below this line, could we rename `self._series` in `self._df` ?

Reply:
I'd also rename the __init__ argument `series` into `df`

Reply:
done